server/src/test/java/org/elasticsearch/index/mapper/BooleanFieldMapperTests.java

```diff
@@ -162,4 +164,20 @@ public void testDocValues() throws Exception {
         assertEquals(DocValuesType.NONE, fields[0].fieldType().docValuesType());
         assertEquals(DocValuesType.SORTED_NUMERIC, fields[1].fieldType().docValuesType());
     }
+
+    @Override
+    protected Supplier<? extends Object> randomFetchTestValueVendor(MappedFieldType ft) {
+        switch (between(0, 3)) {
```

Comment:
This will definitely be nicer when it doesn't return a `Supplier` as then we get a random mix of types within a single doc.